### PR TITLE
Add joker

### PR DESCRIPTION
**Pull request by amilajack**
#### New App Submission

**App name:** Joker

**Repo or homepage link:** https://github.com/amilajack/joker

**Description:** A modern and intuitive testing library for command-line apps

**Why you think it's awesome:** There aren't a lot of CLI end to end testing tools that are well maintained. We have amazing testing frameworks for React (jest, mocha, etc) but we don't have good ones for node CLI's. This project fixes that!

### CODE REVIEW

**Comment by jneidel** [comment]
> There aren't a lot of CLI end to end testing tools that are well maintained.

What about [execa](https://github.com/sindresorhus/execa)?

**Comment by amilajack** [comment]
By command line tools I mean CLI apps. So things like the following: 
* https://github.com/klaussinani/taskbook
* https://github.com/klaussinani/signale
Also I went through some of the apps in the README and many of them don't seem to have tests.

**Comment by jneidel** [comment]
@amilajack What you mean are TUIs. See [this post](https://askubuntu.com/a/867426) for the difference.

**Comment by jneidel** [comment]
From the readme examples it looks a lot like execa though.

Not to put down your work, I just want to know the difference, especially since you claimed that there is no such maintained tool.

How does it test the interactive aspects of, for example [taskbook](https://github.com/klaussinani/taskbook)?

**Comment by amilajack** [comment]
I'd recommend taking a look at the [API docs in the readme](https://github.com/amilajack/joker#features). They show some of the basic testing capabilities and functionality provided by the library

**Comment by jneidel** [comment]
@amilajack  Makes sense, you should add that in a condensed form to the readme. As a features list, or a text with the heading "comparison to similar tools".

File: readme.md [hunk 1]
```diff
@@ -121,6 +121,7 @@ Inspired by the [awesome](https://github.com/sindresorhus/awesome) list thing.
 - [cgasm](https://github.com/bnagy/cgasm) - A tool that gives x86 assembly documentation. It is pronounced "SeekAzzem".
 - [cookiecutter](https://github.com/audreyr/cookiecutter) - Creates projects from cookiecutters (project templates).
 - [dockly](https://github.com/lirantal/dockly) - Interactive CLI for managing Docker containers.
+- [joker](https://github.com/amilajack/joker) - A modern and intuitive testing library for command-line apps
```
Comment (jneidel) [inline readme.md]: As stated in the contribution guidelines:
"Additions should be added to the bottom of the relevant category."

Please move it to the end of the development list, i.e. below grip (line 129).

**Comment by jneidel** [comment]
I've just realized, this is not a CLI, but a node module for building CLIs and thus is not within the scope of this list, right @sjclayton?

From contribution guidelines:
"The aim of the list is to provide a concise list of awesome **CLI tools and apps**."

**Comment by baltzyee** [comment]
But why is not taskbook in this list? It look very awesome and I use it for about a week and love it!
@jneidel Thank you for the awesome list! I have found very useful apps for terminal!

**Comment by jneidel** [comment]
> But why is not taskbook in this list?

@baltzyee feel free to open a pull request adding it :smile:

**Comment by baltzyee** [comment]
@jneidel Done!

**Comment by sjclayton** [comment]
> I've just realized, this is not a CLI, but a node module for building CLIs and thus is not within the scope of this list, right @sjclayton?
> 
> From contribution guidelines:
> "The aim of the list is to provide a concise list of awesome **CLI tools and apps**."

Yeah, I agree -- I don't believe this should be added to the list after all.

**Comment by sjclayton** [comment]
Closing as this doesn't meet the requirements / guidelines to be including on the list.

**Pull request closed without merging**